### PR TITLE
[schema] feat: add pdfData to file plugin schema

### PR DESCRIPTION
**Pull request by talevy17**

### CODE REVIEW

File: packages/ricos-content/web/src/converters/draft/toDraft/convertDraftPluginData.ts [hunk 1]
```diff
@@ -412,6 +412,10 @@ const convertFileData = (data: FileData & FileComponentData) => {
   data.url = url;
   data.id = id || custom;
   updatePrivacyField(data, isPrivate);
+  if (data.pdfData) {
+    const { viewMode } = data.pdfData;
+    viewMode && ((data.pdfData.viewMode as string) = viewMode.toLowerCase());
```
Comment (danarish4) [inline packages/ricos-content/web/src/converters/draft/toDraft/convertDraftPluginData.ts:417]: why this is different in draft? lets do it upper case both in draft&ricos schema

File: packages/ricos-content/web/src/types/mediaUploadTypes.ts [hunk 2]
```diff
@@ -82,10 +82,18 @@ export interface VideoComponentData {
...
+  disablePrint?: boolean;
+  disableDownload?: boolean;
+}
+
 export interface FileComponentData {
   name: string;
   type: string;
   url?: string;
   id?: string;
+  size?: string | number;
```
Comment (danarish4) [inline packages/ricos-content/web/src/types/mediaUploadTypes.ts:96]: what is the size for ?

Comment (talevy17) [reply packages/ricos-content/web/src/types/mediaUploadTypes.ts:96]: we keep file size in our schema but it was missing in the typescript type for some reason

File: packages/ricos-content/web/src/converters/draft/toDraft/convertDraftPluginData.ts (hunk 1 above)
Comment (talevy17) [reply packages/ricos-content/web/src/converters/draft/toDraft/convertDraftPluginData.ts]: done

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_file.proto [hunk 3]
```diff
@@ -6,9 +6,22 @@ import "wix/rich_content/v1/common.proto";
 import "google/protobuf/wrappers.proto";
 
 message FileData {
+  message PDFData {
+    enum ViewModes {
```
Comment (maimonav) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_file.proto:10]: `ViewModes` ==> `ViewMode` :)

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_file.proto [hunk 4]
```diff
@@ -6,9 +6,22 @@ import "wix/rich_content/v1/common.proto";
 import "google/protobuf/wrappers.proto";
 
 message FileData {
+  message PDFData {
```
Comment (maimonav) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_file.proto:9]: This is more of a settings than data, WDYT about `PDFSettings`? @talevy17

File: packages/ricos-content/web/src/types/mediaUploadTypes.ts [hunk 5]
```diff
@@ -82,10 +82,18 @@ export interface VideoComponentData {
 
 export type MediaPrivacy = 'public' | 'private';
 
+export interface PDFData {
+  viewMode?: string;
+  disablePrint?: boolean;
+  disableDownload?: boolean;
+}
+
```
Comment (maimonav) [inline packages/ricos-content/web/src/types/mediaUploadTypes.ts:90]: Can you reuse the types generated from ricos-schema?

Comment (talevy17) [reply packages/ricos-content/web/src/types/mediaUploadTypes.ts:90]: I'll remove it from this PR and check in the usage if we can use the schema types.

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_file.proto (hunk 4 above)
Comment (talevy17) [reply packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_file.proto:9]: I like it, we overuse the term _Data anyways 👍

**Review by maimonav — CHANGES_REQUESTED** [review]
Hey :), please take a look at the comments.
BTW, can you please add a test that checks the conversion fromDraft & toDraft?
Thanks

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_file.proto (hunk 3 above)
Comment (talevy17) [reply packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_file.proto]: done

File: packages/ricos-content/web/src/types/mediaUploadTypes.ts (hunk 5 above)
Comment (maimonav) [reply packages/ricos-content/web/src/types/mediaUploadTypes.ts:92]: Did it work?

File: packages/ricos-content/web/src/types/mediaUploadTypes.ts [hunk 6]
```diff
@@ -82,10 +84,18 @@ export interface VideoComponentData {
 
 export type MediaPrivacy = 'public' | 'private';
 
+export interface PDFData {
```
Comment (maimonav) [inline packages/ricos-content/web/src/types/mediaUploadTypes.ts:87]: PDFData => PDFSettings :)

File: packages/ricos-content/web/src/types/mediaUploadTypes.ts (hunk 5 above)
Comment (talevy17) [reply packages/ricos-content/web/src/types/mediaUploadTypes.ts:92]: didn't break in compile time, so I guess it did

File: packages/ricos-content/web/src/types/mediaUploadTypes.ts (hunk 6 above)
Comment (talevy17) [reply packages/ricos-content/web/src/types/mediaUploadTypes.ts:87]: I'm taking this type from the schema, should have been deleted (:

Comment (maimonav) [reply packages/ricos-content/web/src/types/mediaUploadTypes.ts:87]: Great!

**Review by maimonav — APPROVED** [review]
LGTM 🔥

**Pull request merged**